File: app/src/main/java/com/example/dayo/data/database/AppDatabase.java
package com.example.dayo.data.database;

import androidx.room.Database;
import androidx.room.RoomDatabase;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

@Database(entities = {User.class, Activity.class}, version = 5, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {
    private static volatile AppDatabase instance;

    public abstract UserDao userDao();
    public abstract ActivityDao activityDao();

    private static final int NUMBER_OF_THREADS = 4;
    public static final ExecutorService databaseWriteExecutor =
            Executors.newFixedThreadPool(NUMBER_OF_THREADS);
}

File: app/src/main/java/com/example/dayo/data/database/AppDatabase.kt
package com.example.dayo.data.database

import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.TypeConverters

@Database(entities = [User::class], version = 1)
@TypeConverters(CategoryConverter::class) // Προσθήκη μετατροπέων
abstract class AppDatabase : RoomDatabase() {
    abstract fun userDao(): UserDao
}